pulsar-common/src/main/java/org/apache/pulsar/common/util/URIPreconditions.java

```diff
@@ -67,7 +67,7 @@ public static void checkURIIfPresent(@Nullable String uri,
     public static void checkURIIfPresent(@Nullable String uri,
                                          @Nonnull Predicate<URI> predicate,
                                          @Nullable String errorMessage) throws IllegalArgumentException {
-        if (uri == null) {
+        if (uri == null || uri.length() == 0) {
```

Comment:
> I think one risk here is that it is now possible to create a cluster with all empty cluster urls. I think it might make sense to verify that there is some URL that is not zero length.

Sure, I agreed with you! We must ensure that at least one URL is a non-zero length when creating the cluster.

Do you make a PR to do that? Recently, I don't have time to maintain the Pulsar :(

Reply:
@michaeljmarshall

Reply:
I'll create a GitHub issue and label it as a good one for first time contributors.

Reply:
I created https://github.com/apache/pulsar/issues/19866.

Also, while looking into it, I realized that this PR didn't introduce the possibility of configuring no urls. The issue I created is very low priority.